src/components/CytoscapeLayout/CytoscapeLayout.tsx

```diff
@@ -67,25 +74,33 @@ export default class CytoscapeLayout extends React.Component<CytoscapeLayoutProp
   }
 
   render() {
-    return (
-      <div id="cytoscape-container" style={{ marginRight: '25em' }}>
-        <ReactCytoscape
-          containerID="cy"
-          cyRef={cy => {
-            this.cyRef(cy);
-          }}
-          elements={this.state.elements}
-          style={CytoscapeConfig.getStyles()}
-          cytoscapeOptions={{ wheelSensitivity: 0.1, autounselectify: false }}
-          layout={{
-            name: 'breadthfirst',
-            directed: 'true',
-            maximalAdjustments: 2,
-            spacingFactor: 1
-          }}
-        />
-      </div>
-    );
+    if (this.state.isLoading === true) {
+      return (
+        <div id="cytoscape-container" style={{ marginRight: '25em' }}>
+          <Spinner loading={true} />
```

Comment:
IIRC you can put the ReactCytoscape inside the Spinner and it will hide it when loading=true. With that you can remove the if/else

Reply:
Oh, I didn't know I could do that. That makes things easier, I will update

Reply:
Hmm ... I just ran some tests with this `<Spinner loading={this.state.isLoading}>` and embedded jsx, but it is actually less efficient since all the child JSX is still interpreted, functions calls executed etc. Whereas it doesn't if you check with `if (this.state.isLoading)`.

Reply:
Im curious on how you ran your tests. I did some quick testing and not everything seems to be executed.

e.g.
This doesn't print "hello world" in the console, nor is the DOM element <div>hello world</div> in the tree (not even with hidden or something like that).

```json
<Spinner style={{ marginTop: '15px' }} loading={true} size="lg" inverse={true}>
           <div ref={am => { console.log('hello world'); }}>hello world</div>
          </Spinner>
```

Looking at the code [1] if loading == true, it won't return the children (although i'm not an expert in react, so just guessing).

[1] https://github.com/patternfly/patternfly-react/blob/master/src/components/Spinner/Spinner.js#L20

Reply:
@josejulio I did write something like `{console.log(...)}` in children, not as an anonymous function like you do.
So you're right that it's not hidden, the children dom is really not interpreted. But embedded js code is executed.
Of course since js code must be as minimal as possible in a render function anyway, there shouldn't be much impact - But still it's good to keep that in mind

To summarize: @mwringe you can do as you want :)